pkg/find-interface_test.go

```diff
@@ -0,0 +1,19 @@
+package impl
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFindTopLevel(t *testing.T) {
```

Comment:
Is there a reason you've added so many unit tests? I'd personally just write higher-level tests that run the tool with a Go file input and output. If a regression test cannot be written with a full "integration" test, there likely was no bug to begin with.

Reply:
It's mostly that unit tests are easier to write for progressive code. I wasn't likely going to get everything finished in a single leap, so unit testing is more helpful for adding a bit here and there, and getting that flywheel going, so to speak.

My experience has also been that broken granular unit tests are usually a bit easier to pinpoint the root cause and fix than broken integration suites. Sometimes it's just not obvious where the bug came from even in a single integration test, whereas granular Unit tests usually only involve a few failures that point you right at the cause.